src/actions/mod.rs

```diff
@@ -156,6 +156,23 @@ impl ActionHandler {
         }
     }
 
+    pub fn find_impls(&self, id: usize, params: TextDocumentPositionParams, out: &Output) {
+        let span = self.convert_pos_to_span(&params.text_document, params.position);
+        let type_id = self.analysis.id(&span).unwrap();
+        // FIXME send failure
+        let res = self.analysis
+            .find_impls(type_id)
+            .unwrap_or(vec![])
+            .into_iter()
+            // FIXME: these are part of libstd and libcore but the path probably does not
+            // exists and thus should be removed. Replace with correct path later.
+            .filter(|x| x.file.starts_with(Path::new("/checkout/src/")))
```

Comment:
I'm curious about this - why is it needed here and not for other queries? I think ideally any filtering like this should happen in rls-analysis, not here (and probably in a more principled way). I also think, but not sure, that this is probably incorrect, since this will cover *all* upstream crates, not just core/std, but it is only the std/core ones that do not exist?